pyproject.toml

```diff
@@ -89,3 +86,71 @@ extend-safe-fixes = [
 [tool.pyright]
 venvPath = "."
 venv = "venv"
+
+[tool.pixi.project]
+channels = ["conda-forge"]
+platforms = ["linux-64", "osx-arm64", "osx-64", "win-64"]
+
+[tool.pixi.pypi-dependencies]
+xarray_subset_grid = { path = ".", editable = true }
+
+[tool.pixi.environments]
+default = { solve-group = "default" }
+dev = { features = ["dev"], solve-group = "default" }
+examples = { features = ["examples"], solve-group = "default" }
+all = { features = ["dev", "examples"], solve-group = "default" }
+test39 = ["dev", "py39"] # implicit: test39 = ["dev", "py39", "default"]
+test310 = ["dev", "py310"] # implicit: test39 = ["dev", "py39", "default"]
+test311 = ["dev", "py311"]
+test312 = ["dev", "py312"]
+
+
+[tool.pixi.tasks]
+
+[tool.pixi.dependencies]
```

Comment:
Do pixi dependencies need to be listed in addition to the package dependencies? Would love not to have to duplicate

Reply:
They do, because there is not a 1:1 relationship between PyPi and conda-forge packages. There are ways to figure out the mapping (grayskull does it) but I don't know how robust it is, and I don't think pixi supports it anyway.

I'm still wrapping my head around what pixi is for, exactly. But I *think* it's more a "Project" environment manager. So the pixi requirements are really about what you need to do a project with this package -- e.g. develop it, test it, run the examples, etc, rather than what you need to install the package and use it somewhere else.

The requirements in the main part of the pyproject.toml are for putting in with a built package to distribute it. (and what you need to build it).

The conda version of that is the requirements in a conda recipe -- which we may get to once this stabilizes ... (then there will be three copies of the requirements .. sigh)

One thing I think we could simplify is the three pixi features -- do we really need base, and dev and examples? We could simply put the whole thing in base -- or maybe all the dev dependencies in base, and then add an examples one (or not have that at all ...) ...

Reply:
Okay then im good with how it is for now then! Thank you for the long explanation